# Task 2

## Answer 2.a

Using the definition of marginal VaR,

$$\frac{\partial VaR(\omega)}{\partial \omega_i}=\mu_i+z_\alpha \frac{(\Sigma\omega)_i}{\sqrt{\omega^{\top}\Sigma\omega}}$$

the result can be verified by computing the marginal VaR vector and comparing it with the values stated in the question.

$$\text{marginal VaR vector} = \mu + z_\alpha \frac{\Sigma\omega}{\sqrt{\omega^{\top}\Sigma\omega}}$$

The calculation therefore begins with $\Sigma\omega$ and $\sqrt{\omega^{\top}\Sigma\omega}$.

The quantities provided in the question are defined below.

$\mu$ vector:

$$
\mu=\begin{pmatrix}
0 \\
0 \\
0 \\
\end{pmatrix}
$$

$\sigma$ vector:

$$
\sigma=\begin{pmatrix}
0.30 \\
0.20 \\
0.15 \\
\end{pmatrix}
$$

$$
S=\begin{pmatrix}
0.30 & 0 & 0\\
0 & 0.20 & 0 \\
0 & 0 & 0.15 \\
\end{pmatrix}
$$

$\omega$ vector:

$$
\omega=\begin{pmatrix}
0.50 \\
0.20 \\
0.30 \\
\end{pmatrix}
$$

$\rho$ matrix:

$$
R=\begin{pmatrix}
1.0 & 0.8 & 0.5 \\
0.8 & 1.0 & 0.3 \\
0.5 & 0.3 & 1.0 \\
\end{pmatrix}
$$

In [2]:
import numpy as np

mu = np.array([[0],
               [0],
               [0]])

sd = np.array([[0.30],
               [0.20],
               [0.15]])

S = np.diag(sd.flatten())

w = np.array([[0.5],
              [0.2],
              [0.3]])

R = np.array([[1.0, 0.8, 0.5],
              [0.8, 1.0, 0.3],
              [0.5, 0.3, 1.0]])

This gives
$$\Sigma = SRS^{\top} = SRS =
\begin{pmatrix}
0.0900	& 0.0480	& 0.0225 \\
0.0480	& 0.0400	& 0.0090 \\
0.0225	& 0.0090	& 0.0225 \\
\end{pmatrix}
$$

In [3]:
SRS = S @ R @ S
SRS

array([[0.09  , 0.048 , 0.0225],
       [0.048 , 0.04  , 0.009 ],
       [0.0225, 0.009 , 0.0225]])

Hence,

$$\Sigma\omega=
\begin{pmatrix}
0.0900	& 0.0480	& 0.0225 \\
0.0480	& 0.0400	& 0.0090 \\
0.0225	& 0.0090	& 0.0225 \\
\end{pmatrix}
\begin{pmatrix}
0.5 \\
0.2 \\
0.3 \\
\end{pmatrix}
=\begin{pmatrix}
0.0614 \\
0.0347 \\
0.0198 \\
\end{pmatrix}
$$

In [5]:
sigma_w = SRS @ w
sigma_w

array([[0.06135],
       [0.0347 ],
       [0.0198 ]])

Moreover,

$$
\omega^{\top}\Sigma\omega=
\begin{pmatrix}
0.5 \\
0.2 \\
0.3 \\
\end{pmatrix}^{\top}
\begin{pmatrix}
0.0900	& 0.0480	& 0.0225 \\
0.0480	& 0.0400	& 0.0090 \\
0.0225	& 0.0090	& 0.0225 \\
\end{pmatrix}
\begin{pmatrix}
0.5 \\
0.2 \\
0.3 \\
\end{pmatrix}
=0.043555
$$

$$\sqrt{\omega^{\top}\Sigma\omega}=0.20869835$$

In [6]:
w_sigma_w = w.T @ sigma_w
w_sigma_w_sqrt = np.sqrt(w_sigma_w)
print(f'w_sigma_w: {w_sigma_w}, w_sigma_w_sqrt: {w_sigma_w_sqrt}')

w_sigma_w: [[0.043555]], w_sigma_w_sqrt: [[0.20869835]]


Finally, we compute the marginal VaR vector:

$$\mu + z_\alpha \frac{\Sigma\omega}{\sqrt{\omega^{\top}\Sigma\omega}}$$

In [76]:
z = -2.326
sensitivity = mu + z * sigma_w / w_sigma_w_sqrt
sensitivity

array([[-0.68376248],
       [-0.38674096],
       [-0.2206764 ]])

Hence, the three marginal VaR sensitivities are

$$\frac{\partial VaR}{\partial \omega_1} = -0.68376248$$
$$\frac{\partial VaR}{\partial \omega_2} = -0.38674096$$
$$\frac{\partial VaR}{\partial \omega_3} = -0.22067640$$

These values agree with those given in the question.

---


## Answer 2.b
1. All three assets are positively correlated with the portfolio, so increasing the weight of any asset increases portfolio volatility and tail risk.
Since $z_\alpha < 0$ and the expected returns are zero, the risk term dominates, so the marginal VaR and marginal ES sensitivities are negative.

2. A negative sensitivity means that an increase in the weight of that asset makes portfolio VaR more negative, implying a larger potential loss and therefore greater portfolio risk.


---


## Answer 2.c
Asset 1 has the largest value of $(\Sigma\omega)_i$, reflecting both its relatively high volatility and its strong contribution to overall portfolio risk. Therefore, it has the most negative marginal VaR sensitivity.

---

## Answer 2.d
### Mathematical reason

Since $\mu_i = 0$, it is sufficient to compare the absolute values of $|z_\alpha|$ and $\left|\frac{\phi(z_\alpha)}{1-c}\right|$.

Since

$$|z_\alpha| = 2.326$$

$$\left|\frac{\phi(z_\alpha)}{1-c}\right| = \left|\frac{0.02667}{0.01}\right| = 2.667$$

Thus

$$\left|\frac{\phi(z_\alpha)}{1-c}\right| > |z_\alpha|$$

Therefore,

$$\left|\frac{\partial ES}{\partial \omega_i}\right| > \left|\frac{\partial VaR}{\partial \omega_i}\right|$$


### Definitions of VaR and ES
VaR only captures the quantile threshold, whereas ES measures the average loss beyond that threshold. ES is therefore more sensitive to tail risk.

Therefore, changes in marginal weights have a larger impact on ES.